### DISCUSSION TITLE
When testcases fail,  don't display too long logs

### DISCUSSION THREAD

**Discussion by rgbygv** [j178/leetgo#242] [Ideas]
When come to big testcases, too long input and output is useless, just give the wrong info enough.

**Comment by j178** [comment]
Do you have any examples?

**Comment by rgbygv** [comment]
When it comes big testcases, teiminal will be flushd by testcases
But it's so big that it's almost impossible to debug, so i think it's useless info.

```
root@vthe ~/l/contest (main)# lc test last/4 -Ls
● running test locally question=find-building-where-alice-and-bob-can-meet
● Case 1:    Passed
● Case 2:    Passed
● Case 3:    Passed
● Case 4:    Passed
● Case 5:    Passed
● submitting solution

 × Time Limit Exceeded
                      
Passed cases:  946/949
Last case:     [999999999,1,3,6,7,7,10,11,13,16,17,17,20,25,25,27,30,32,35,38,42,43,43,45,46,49,52,53,55,59,61,62,64,67,69,71,76,78,79,81,84,84,84,84,85,86,87,87,91,93,99,101,103,114,116,117,119,121,122,122,124,125,125,125,132,136,136,140,145,147,148,153,154,154,157,158,159,159,161,162,164,164,164,166,167,171,171,179,184,184,186,187,200,201,202,203,204,207,207,208,212,214,214,215,215,216,220,221,228,235,235,236,240,242,242,244,247,250,252,256,256,259,261,263,264,264,266,271,272,274,276,280,282,282,285,289,289,296,299,303,310,311,313,314,315,315,317,319,321
...
hundreds of lines
...
```
since testcases is too long , I can't paste all.
here is code
```
class SegmentTree:
    def __init__(self, nums):
        n = len(nums)
        # f记录的是特定区间，f[k]，序号为k的点：该节点掌管的索引为l,r，值区间l~r的数字总和
        self.nums = [0] + nums  # 加一个哨兵节点，使得数组的有效索引为1～n
        self.f = [0 for _ in range(4 * n)]
        self.buildTree(1, 1, n)

    def buildTree(self, k, l, r):
        # 序号为k的索引，掌管的范围是l~r
        # 这里要注意，对于一棵数组长度确定的线段树，k是可以唯一确定l,r的
        # 例如根节点1 一定对应 1~n
        # 即同一个k对应唯一的l,r
        if l == r:
            # 叶子节点
            self.f[k] = self.nums[l]
            return
        mid = (l + r) // 2
        # 分治 + 后序遍历的思想
        self.buildTree(2 * k, l, mid)  # 处理左孩子
        self.buildTree(2 * k + 1, mid + 1, r)  # 处理右孩子
        self.f[k] = max(self.f[2 * k], self.f[2 * k + 1])  # 父节点的信息为左右孩子汇总

    def update(self, k, l, r, i, x):
        # 序号为k的索引，掌管的范围是l~r
        # 对数组索引为i的节点增量更新x
        self.f[k] += x
        if l == r:
            # 叶子节点
            return
        mid = (l + r) // 2
        # 看索引i在左右子树的哪一边。递归更新
        if i <= mid:  # 在左子树
            self.update(2 * k, l, mid, i, x)
        elif i > mid:  # 在右子树
            self.update(2 * k + 1, mid + 1, r, i, x)

    # @cache
    def query(self, k, l, r, start, end):
        # start~end始终是l~r的子区间
        # 序号为k的索引，掌管的范围是l~r
        # 在整棵树上进行搜寻 start~end 索引所汇总的范围和
        if l == start and r == end:
            return self.f[k]
        mid = (l + r) // 2
        if end <= mid:  # 如果start~end完全在左半边，则只需要算左子树
            return self.query(2 * k, l, mid, start, end)
        if mid < start:  # 如果start~end完全在右半边，则只需要算右子树
            return self.query(2 * k + 1, mid + 1, r, start, end)
        # 否则，需要同时考虑左右孩子
        leftPart = self.query(2 * k, l, mid, start, mid)  # 注意：在这里最后一个参数是mid而不是end
        rightPart = self.query(
            2 * k + 1, mid + 1, r, mid + 1, end
        )  # 注意：在这里倒数第二个参数是mid+1而不是start
        # 因为：# start~end始终是l~r的子区间，否则递归会没有出口
        return max(leftPart, rightPart)


class Solution:
    def leftmostBuildingQueries(
        self, heights: List[int], queries: List[List[int]]
    ) -> List[int]:
        n = len(heights)
        st = SegmentTree(heights)
        res = []

        for x, y in queries:
            if x > y:
                x, y = y, x
            if heights[y] > heights[x] or x == y:
                res.append(y)
                continue
            hx = max(heights[x], heights[y])
            l = y + 1
            r = n - 1
            while l < r:
                mid = (l + r) >> 1
                cur = st.query(1, 1, n, l + 1, mid + 1)
                # print(l, r, cur)
                if cur > hx:
                    r = mid
                else:
                    l = mid + 1
            if l >= n or heights[l] <= hx:
                res.append(-1)
            else:
                res.append(l)
        # st.query.cache_clear()
        return res

```

**Comment by j178** [comment]
Thanks, we should put a max length limit on the input/output, do you have any interest to contribute? 😄

**Comment by rgbygv** [comment]
I may not know what the appropriate size is. 
I have encountered situations where excessive output has filled up the terminal a few times, so I think there should be a limit on the output.
I will inform you again if I have good ideas.

**Comment by j178** [comment]
Thanks. I just merged #244, welcome feedback.

**Comment by rgbygv** [comment]
I found that a limit of 100 might be too small for python. Because the length of general questions is very long, plus python’s traceback, it may be hard to know the error message.

Maybe setting the limit to 1000 would be better. But I'm not sure if this is a good limitation for other languages.
Or we should only truncate wrong answers , not other errors.

leetgo output

![Code_VMPdWiEoq3](https://github.com/j178/leetgo/assets/49787317/b507eb99-cd43-4c8c-aa02-0f2e00b35eba)

local output
![Code_iFBixshS2V](https://github.com/j178/leetgo/assets/49787317/1c948288-dfab-4d7c-9657-4779824eb669)

**Reply by j178** [reply]
I changed the limit of stdout/stderr length to 1000 at ~https://github.com/j178/leetgo/commit/7c3c646e9cac2fd25e18bcd15eda278248fab43c~ https://github.com/j178/leetgo/commit/4f444acb69110c7e1d39d5f073376454d4c1134f, could you try it?

**Reply by rgbygv** [reply]
Thanks for quick reply, but still have some problem. It seems like there are some newlines connecting the output

leetgo output 
![Code_oAAZy0y2kt](https://github.com/j178/leetgo/assets/49787317/8bdc4ef7-175f-45c6-964d-a0d82e86aa8a)

local output
![Code_tyXAatlkiw](https://github.com/j178/leetgo/assets/49787317/b04058fc-3c6a-4063-afde-8410687aebf0)

Other parts work normally
![Code_TP7H2TUkcg](https://github.com/j178/leetgo/assets/49787317/56a82545-584d-4828-8dae-ccda8dbf81e5)

![Code_N0XEd6Rzg0](https://github.com/j178/leetgo/assets/49787317/ce5ce70d-92db-4495-84f4-04e549aebb0a)

**Reply by j178** [reply]
Sorry, but I cannot reproduce this on my computer. Maybe the [`SplitLines`](https://github.com/j178/leetgo/blob/4f444acb69110c7e1d39d5f073376454d4c1134f/utils/str.go#L28) function is not working properly under your environment.

**Reply by rgbygv** [reply]
You are right.
I tried using the same code in another question, but I couldn't reproduce it either.
Sorry for the misunderstanding.
